bin/graphics-core22-cleanup

```diff
@@ -3,6 +3,32 @@ set -euo pipefail
 
 SELF="$( cd -- "$(dirname "$0")/.." ; pwd -P )"
 
+TMP_FOLDER=/tmp/snap-lists
+
+rm -rf ${TMP_FOLDER}
+mkdir -p ${TMP_FOLDER}
+
+(
+  for provider in "$@"; do
+    snap download --target-directory ${TMP_FOLDER}/ --basename ${provider} ${provider}
+    # Process the list exactly like the CI script, to allow to compare it
+    # with the ones here.
+    unsquashfs -cat ${TMP_FOLDER}/${provider}.snap snap/${CRAFT_TARGET_ARCH}.list \
+             | sed 's/\(.so.[0-9]\+\)\([0-9\.]\+\)\?$/\1*/' \
+             | sort -u \
+             > ${TMP_FOLDER}/original-list.txt
+    sort -u ${SELF}/lists/${provider}.${CRAFT_TARGET_ARCH}.list \
+             > ${TMP_FOLDER}/compare-list.txt
+    # If the lists are different, cmp will fail and thus the script too.
+    # This ensures that if the upstream snap and this one aren't in sync,
+    # the build will be aborted
+    cmp ${TMP_FOLDER}/original-list.txt ${TMP_FOLDER}/compare-list.txt
```

Comment:
Maybe echo an error pointing at https://github.com/canonical/gpu-snap/issues? If this fails we want to explain why.

```suggestion
    cmp ${TMP_FOLDER}/original-list.txt ${SELF}/lists/${provider}.${CRAFT_TARGET_ARCH}.list
```